src/Fixer/Import/GlobalNamespaceImportFixer.php

```diff
@@ -707,7 +707,7 @@ private function traverseDocBlockTypes(DocBlock $doc, callable $callback): bool
         foreach ($annotations as $annotation) {
             $types = $new = $annotation->getTypes();
 
-            foreach ($types as $i => $fullType) {
+            foreach ($new as $i => $fullType) {
```

Comment:
why the change?
to what i recall, idea was to iterate over _old_ ($types) array and put modifications to _new_ ($new) array, and compare if there were any differences

Reply:
At this line `$types` and `$new` are the same, so the behavior did not change.

But when iterating over `$new` phpstan is able to understand that we only edit existing keys in line 723, not adding new keys. So phpstan is sure that `$new` is still a `list` after the `foreach` loop.

So this avoids this error in line 727:

> Parameter `#1` $types of method PhpCsFixer\\DocBlock\\Annotation::setTypes() expects list<string>, array<int<0, max>, string> given.

Reply:
Not sure it is really a bug.  
https://phpstan.org/r/2ad3cfd0-13cd-4d76-9a35-8c0b0d9c2dce

I would not expect that phpstan knows inside `foreach` loop, that all keys of `$a` also already exist in `$b`.

Reply:
array_map ? :)